clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1881,15 +1941,24 @@ scheduler_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t output,
                 // We're on the last record.
             } else if (outputs_[output].record[outputs_[output].record_index].type ==
                        schedule_record_t::SKIP) {
+                VPRINT(this, 5, "next_record[%d]: need new input after skip\n", output);
                 need_new_input = true;
             } else {
-                uint64_t stop = outputs_[output]
-                                    .record[outputs_[output].record_index]
-                                    .stop_instruction;
+                const schedule_record_t &segment =
+                    outputs_[output].record[outputs_[output].record_index];
+                uint64_t start = segment.start_instruction;
+                uint64_t stop = segment.stop_instruction;
                 // The stop is exclusive.  0 does mean to do nothing (easiest
                 // to have an empty record to share the next-entry for a start skip
                 // or other cases).
-                if (input->reader->get_instruction_ordinal() >= stop) {
+                // Only check for stop when we've exhausted the queue, or we have
+                // a starter schedule with a 0,0 entry prior to a first skip entry.
+                if (input->reader->get_instruction_ordinal() >= stop &&
+                    (!from_queue || (start == 0 && stop == 0))) {
```

Comment:
Can you describe what this starter 0,0 entry means in a stored schedule file?

Reply:
It was easier to have this seemingly-redundant entry to get into the trace reading loop and then do something like a skip from the start rather than adding logic into the setup code to do an initial start on replay.  Elaborated in the comment here.